pkg/ipam/metal3io/metal3io_ipam.go

```diff
@@ -109,8 +109,10 @@ func (m Metal3IPAM) GetAvailableIPPool(poolMatchLabels map[string]string, cluste
 
 	//TODO: refactor searchDomains, once its added in metal3io
 	searchDomains := []string{}
-	if sdList, ok := ipPool.Annotations[ipam.SearchDomainsKey]; ok {
-		searchDomains = strings.Split(sdList, ",")
+	if len(ipPool.Annotations[ipam.SearchDomainsKey]) > 0 {
+		if sdList, ok := ipPool.Annotations[ipam.SearchDomainsKey]; ok {
```

Comment:
is this if check needed when we have the line above?

Reply:
This is to skip the case where the annotation has empty string ("") as the value.

Reply:
line 112 already check for the length, so that should already cover the empty case?

Reply:
Ah, you were pointing to line 113. Ack, will update.